src/conversion/mapper.js

```diff
@@ -159,13 +159,12 @@ export default class Mapper {
 	toModelPosition( viewPosition ) {
 		const data = {
 			viewPosition: viewPosition,
-			modelPosition: this._defaultToModelPosition( viewPosition ),
 			mapper: this
 		};
 
 		this.fire( 'viewToModelPosition', data );
 
-		return data.modelPosition;
+		return data.modelPosition ? data.modelPosition : this._defaultToModelPosition( viewPosition );
```

Comment:
So, you mean something that resembles real case scenario. Well, the problem with such tests is that you need to think of some non-dumb case/feature and then code it. So you either go easy way (like the linked tests) or start re-writing other feature (i.e. List). Maybe it would be better to drop some reference in the event description (to List feature?).